migration/app_state.go

```
@@ -0,0 +1,450 @@
+		addresses[address] = struct{}{}
+		if _, duplicate := accountNumbers[account.GetAccountNumber()]; duplicate {
+			return nil, nil, fmt.Errorf(
+				"migration: duplicate auth account number %d",
+				account.GetAccountNumber(),
+			)
+		}
+		accountNumbers[account.GetAccountNumber()] = struct{}{}
+		if _, target := targets[address]; target {
+			return nil, nil, fmt.Errorf("migration: replacement auth account %q already exists", address)
+		}
+		if _, module := account.(authtypes.ModuleAccountI); module {
```

golangci-lint (2.12.2)

[error] SA1019: authtypes.ModuleAccountI is deprecated: Use `ModuleAccountI` from types package instead. 

(staticcheck)